protostar/commands/test/test_result_structured_formatter.py

```diff
@@ -0,0 +1,224 @@
+from typing import Any, Literal, Optional
+from dataclasses import dataclass
+from pathlib import Path
+
+from protostar.protostar_exception import UNEXPECTED_PROTOSTAR_ERROR_MSG
+from protostar.testing import (
+    BrokenFuzzTestCaseResult,
+    BrokenTestCaseResult,
+    BrokenTestSuiteResult,
+    FailedFuzzTestCaseResult,
+    FailedTestCaseResult,
+    PassedFuzzTestCaseResult,
+    PassedTestCaseResult,
+    SkippedTestCaseResult,
+    TestResult,
+    UnexpectedBrokenTestSuiteResult,
+)
+from protostar.io import StructuredMessage, LogColorProvider
+from protostar.testing import OutputName
+
+JsonData = dict[str, Any]
+
+
+@dataclass
+# pylint: disable=too-many-instance-attributes
+class TestCaseResult(StructuredMessage):
+    status: Literal["passed", "failed", "broken", "skipped", "unexpected_exception"]
+    test_suite_path: Path
+
+    test_case_name: Optional[str] = None
+    execution_time_in_seconds: Optional[float] = None
+    exception: Optional[str] = None
+    stdout: Optional[dict[OutputName, str]] = None
+    traceback: Optional[str] = None
+    protostar_message: Optional[str] = None
+
+    fuzz_runs: Optional[int] = None
+    gas: Optional[str] = None
+    steps: Optional[str] = None
+    memory_holes: Optional[str] = None
+    builtins: Optional[list[JsonData]] = None
+
+    reason: Optional[str] = None
+
+    type: str = "test_case_result"
+
+    def format_human(self, fmt: LogColorProvider) -> str:
+        assert False, "Tests should use live logging for the human-readable output"
```

Comment:
This method should be implemented. Instead of creating one `UberTestCaseResultMessage` I'd create a message for each testing result (or formatter function). `LiveLogger` should use messages. I'd create a new directory `protostar/commands/test/messages` and keep each message in a separate file.

Reply:
I understand this comment as two comments actually that are unrelated to each other:

1.
> This method should be implemented.

As I understand, you referred to `format_human`. Why do you think it should be implemented? This part of the functionality stands as the opposite of the human-readable output and should print only JSON, right? Can you specify what output would be expected here and why should we implement this?

2. 

> Instead of creating one `UberTestCaseResultMessage` I'd create a message for each testing result (or formatter function). `LiveLogger` should use messages. I'd create a new directory `protostar/commands/test/messages` and keep each message in a separate file.

This is something totally different and I agree that we can create a hierarchy here and split the message class into multiple smaller message classes, one for each type of test result.

Reply:
> As I understand, you referred to format_human. Why do you think it should be implemented?

It's part of the Marek's design. That's why `StructuredMessage` forces you to implement `format_human` and this code fights against the design constraints, which results in:
- code duplication
  - `format_test_result_structured`
  - `json.dumps` in `TestingLiveLogger` is already implemented in `JsonMessenger` which I've just noticed is configured differently than `json.dumps` in `TestingLiveLogger`. 
- lack of consistency with the way how the `StructuredMessage` is used by other commands.

> This part of the functionality stands as the opposite of the human-readable output and should print only JSON, right? Can you specify what output would be expected here and why should we implement this?

`format_human` should return the same thing what `format_test_result` returns.

Reply:
Please check out the newest changes